src/gtfs/native/Association.ts

```diff
@@ -92,6 +92,11 @@ export class Association implements OverlayRecord {
       end = assoc.after(this.assocLocation);
     }
     else {
+      // this should never happen, unless data feed is corrupted. It will prevent us from update failure (see: JP-404).
+      if (assoc.stopAt(this.assocLocation) === undefined || base.stopAt(this.assocLocation) === undefined) {
+        return assoc;
```

Comment:
I think you should move it to line 86 to cover both association types, extract the two stops into variables and add a log message of some kind (i.e. move this method outside of the entity).

Reply:
Adding a log message will not be easy, but at least do the other two please :)